src/SeleniumLibrary/keywords/screenshot.py

```diff
@@ -14,174 +14,108 @@
-        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
-        | Should Be Equal | ${file1} | ${OUTPUTDIR}${/}selenium-screenshot-1.png |
-        | ${file2} = | Capture Page Screenshot |
-        | File Should Exist | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
-        | Should Be Equal | ${file2} | ${OUTPUTDIR}${/}selenium-screenshot-2.png |
-
-        Example 2:
-        | ${file1} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
-        | ${file2} = | Capture Page Screenshot | ${OTHER_DIR}${/}some-other-name-{index}.png |
-        | ${file3} = | Capture Page Screenshot | ${OTHER_DIR}${/}other-{index}-name.png |
-        | File Should Exist | ${OTHER_DIR}${/}other-1-name.png |
-        | Should Be Equal | ${file1} | ${OTHER_DIR}${/}other-1-name.png |
-        | File Should Exist | ${OTHER_DIR}${/}some-other-name-1.png |
-        | Should Be Equal | ${file2} | ${OTHER_DIR}${/}some-other-name-1.png |
-        | File Should Exist | ${OTHER_DIR}${/}other-2-name.png |
-        | Should Be Equal | ${file3} | ${OTHER_DIR}${/}other-2-name.png |
-
-        Example 3:
-        | Capture Page Screenshot | ${OTHER_DIR}${/}sc-{index:06}.png |
-        | File Should Exist | ${OTHER_DIR}${/}sc-000001.png |
+        ``persist`` argument is deprecated and has no effect.
+
+        The previous value is returned and can be used to restore
+        the original value later if needed.
 
+        Deprecating ``persist`` and returning the previous value are new
+        in SeleniumLibrary 3.0.
         """
-        try:
-            self.browser
-            
-        except RuntimeError:
-            self.info("Couldn't capture page screenshot because no browser is opened")
-        
+        if is_noney(path):
+            path = None
         else:
-            path, link = self._get_screenshot_paths(filename)
+            path = os.path.abspath(path)
             self._create_directory(path)
-            if hasattr(self.browser, 'get_screenshot_as_file'):
-                if not self.browser.get_screenshot_as_file(path):
-                    raise RuntimeError('Failed to save screenshot ' + link)
-            else:
-                if not self.browser.save_screenshot(path):
-                    raise RuntimeError('Failed to save screenshot ' + link)
-            # Image is shown on its own row and thus prev row is closed on purpose
-            msg = (
-                '</td></tr><tr><td colspan="3"><a href="{}">'
-                '<img src="{}" width="800px"></a>'.format(link, link)
-            )
-            self.info(msg, html=True)
-            return path
+        if persist != 'DEPRECATED':
+            self.warn("'persist' argument to 'Set Screenshot Directory' "
+                      "keyword is deprecated and has no effect.")
+        previous = self.ctx.screenshot_root_directory
+        self.ctx.screenshot_root_directory = path
+        return previous
 
     def _create_directory(self, path):
```

Comment:
The method is used also in capture_page_screenshot method. It would be cleaner to move the _create_director method after the capture_page_screenshot methods.

Reply:
This is highly subjective. There's no rule to dictate how to order (helper) methods that would work well in all cases. A common approach, and also approach I generally prefer, is to have helper methods right after their first usage. I don't feel strongly about it here, though.

Reply:
It's highly subjective. The general style, before my time, with the library has been the helpers methods go after the keyword methods. Should we change the style?

Reply:
Having all helper methods at the end in somewhat random order is, in my opinion, a bad practice. I like the stepdown rule recommended in the Clean Code book by Robert C. Martin: https://stackoverflow.com/questions/21881605/resharper-and-stylecop-vs-the-step-down-rule-clean-code

Even with that style there's a question how to handle same helper used multiple times. I prefer having the helper after the first usage, but here it's recommended to have it after the last: http://tidyjava.com/stepdown-rule/

Anyway, I don't think it is a good idea to do larger changes to the code just to change the style, but I'd be in favor of moving towards the stepdown rule. Perhaps having helpers after the last usage would be best, because it would be closer to the current approach.

Notice also that lot of this problem goes away when classes aren't too big. Some of the internal classes could still be split to smaller pieces.